src/TesApi.Web/AzureProxy.cs

```diff
@@ -245,15 +250,15 @@ public async Task<AzureBatchAccountQuotas> GetBatchAccountQuotasAsync()
         public async Task CreateBatchJobAsync(string jobId, CloudTask cloudTask, PoolInformation poolInformation)
         {
             var job = batchClient.JobOperations.CreateJob(jobId, poolInformation);
+            job.OnAllTasksComplete = OnAllTasksComplete.TerminateJob;
             await job.CommitAsync();
 
             try
             {
-                job = await batchClient.JobOperations.GetJobAsync(job.Id); // Retrieve the "bound" version of the job
-                job.PoolInformation = poolInformation;  // Redoing this since the container registry password is not retrieved by GetJobAsync()
-                job.OnAllTasksComplete = OnAllTasksComplete.TerminateJob;
+                job = await batchRaceConditionJobNotFoundRetryPolicy.Execute(async () =>
+                    await batchClient.JobOperations.GetJobAsync(job.Id));
```

Comment:
This becomes:
 job = await batchRaceConditionJobNotFoundRetryPolicy.ExecuteAsync(() => batchClient.JobOperations.GetJobAsync(job.Id));